### PR TITLE
Adapt to CommonJS compatibe style

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -144,7 +153,7 @@ Note:
 - If the parameter description just repeats the parameter name, leave it out.
 - If the parameter is `options` it doesn't need a description.
 - If the function returns `void` or a wrapped `void` like `Promise<void>`, leave out `@returns`.
-- If you include an `@example`, there should be a newline above it. There also needs be a newline after the `@example` tag because of [this TypeScript bug](https://github.com/Microsoft/TypeScript/issues/15749).
+- If you include an `@example`, there should be a newline above it. The example itself should be wrapped with triple ticks (```` ``` ````).
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
- If you include an `@example`, there should be a newline above it. The example itself should be wrapped with triple backticks (```` ``` ````).
```

File: readme.md [hunk 2]
```diff
@@ -159,13 +168,13 @@ Example:
...
 
-expectType<string>(await delay(200, {value: '🦄'}));
-expectType<number>(await delay(200, {value: 0}));
+expectType<Promise<string>>(delay(200, {value: '🦄'}));
+expectType<Promise<number>>(delay(200, {value: 0}));
 
-expectType<never>(await delay.reject(200, {value: '🦄'}));
-expectType<never>(await delay.reject(200, {value: 0}));
+expectType<Promise<never>>(delay.reject(200, {value: '🦄'}));
+expectType<Promise<never>>(delay.reject(200, {value: 0}));
```
Comment (sindresorhus) [inline readme.md:180]: Can you add a note about not using `await` and why? I don't think that's obvious to people even if they see Promise used here. It was at least not obvious to me.

File: readme.md [hunk 3]
```diff
@@ -84,51 +85,59 @@ Exported definitions should be documented with [TSDoc](https://github.com/Micros
 Example:
 
 ```ts
-export interface Options {
-	/**
-	Allow negative numbers.
+/// <reference lib="dom"/>
```
Comment (sindresorhus) [inline readme.md:91]: Can you add a note about when to include this, when to include the `node` reference, and when to include any third-party reference?

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Can you mention using `as const` whenever possible too? With a link to the docs about it.

File: readme.md [hunk 4]
```diff
@@ -183,3 +186,5 @@ Note:
 
 - The test file should be named `index.test-d.ts`.
 - `tsd` supports top-level `await`.
+- When testing promise-returning functions don't use the `await` keyword. Instead, assert directly for a `Promise` like in the example above. When you use `await` your function can potentially return a bare value without being wrapped in a `Promise`, `await` will happily accept non-`Promise` values and your test is rendered meaningless.
+- Use [`const` assertions](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#const-assertions) when you need to pass literal or readonly typed values to functions in your tests.
```
Comment (sindresorhus) [inline readme.md:190]: I didn't mean this specifically for testing. I meant it for the type definitions themselves.

File: readme.md [hunk 5]
```diff
@@ -157,6 +157,9 @@ Note:
 - If the API accepts an options-object, define an `Options` interface as seen above. Document default option values using the [`@default` tag](http://usejsdoc.org/tags-default.html) (since interfaces cannot have default values). If the default needs to be a description instead of a basic value, use the formatting `Default: Lorem Ipsum.`.
 - Use `@returns`, not `@return`.
 - Ambient declarations can't have default parameters, so in the case of a default method parameter, document it in the parameter docs instead, as seen in the above example.
+- When you have to use node types install the node types as a dev dependency and add the `/// <reference types="node"/>` triple-slash reference to the top of the file.
+- When you have to use DOM types (`Window`, `Document`, ...) add the `/// <reference lib="dom"/>` triple-slash reference to the top of the file.
+- Third-party library types must be installed as direct dependencies, if required. They usually only require a triple-slash reference when they expose interfaces only in the global namespace. Otherwise prefer imports over triple-slash references.
```
Comment (sindresorhus) [inline readme.md]: These are in the wrong section. They should be in the https://github.com/sindresorhus/typescript-definition-style-guide#checklist section.

File: readme.md (hunk 4 above)
Comment (BendingBender) [reply readme.md:190]: Can you give me an example of how you would use `const` assertions in a definition file?

Comment (sindresorhus) [reply readme.md:190]: Hmm, you're right. I can't think of any. I always forget how limiting the ambient declaration files are...

**Comment by sindresorhus** [comment]
Thanks for doing this :)

**Pull request merged**